lib/streamlit/runtime/caching/cache_utils.py

```diff
@@ -303,10 +308,22 @@ def _handle_cache_miss(
                     return_value=computed_value, func=self._info.func
                 )
 
-    def clear(self):
-        """Clear the wrapped function's associated cache."""
+    def clear(self, *args, **kwargs):
+        """Clear the wrapped function's associated cache.
+        If no arguments are passed, clear the cache of all values.
+        If args/kwargs are provided, clear the cached value for these arguments only."""
         cache = self._info.get_function_cache(self._function_key)
-        cache.clear()
+        if args or kwargs:
+            key = _make_value_key(
+                cache_type=self._info.cache_type,
+                func=self._info.func,
+                func_args=args,
+                func_kwargs=kwargs,
+                hash_funcs=self._info.hash_funcs,
+            )
+        else:
+            key = None
+        cache.clear(key=key)
```

Comment:
nit: we might want to consider catching the exception if the key cannot be found in the cache, and rethrow it with a StreamlitAPIException that has a better description of what is going wrong. I think the key shown in the other exception might not be super descriptive. And maybe an additional test for this case (there isn't a cached entry for the arguments) might make sense.

Reply:
I thought a bit more about this: maybe we shouldn't even throw an exception here if there isn't an item associated. We just `except` and `pass` in that case. If there isn't an item in the cache for the arguments, the user shouldn't even care since there is nothing to delete.